pxtlib/service.ts

```diff
@@ -688,6 +692,21 @@ namespace ts.pxtc {
         return apis;
     }
 
+    function hasEquivalentParameters(a: pxt.blocks.BlockCompileInfo, b: pxt.blocks.BlockCompileInfo) {
+        if (a.parameters.length != b.parameters.length)
+            return false;
+
+        for (const aParam of a.parameters) {
+            const bParam = b.actualNameToParam[aParam.actualName];
+            if (!bParam
+                || aParam.type != bParam.type
+                || aParam.shadowBlockId != bParam.shadowBlockId) {
+                return false;
```

Comment:
add ``pxt.debug`` logging, it's very helpful to pin point exctaly what happened here.